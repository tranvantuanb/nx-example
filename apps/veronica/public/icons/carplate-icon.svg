<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M15 12.3514H1V4H15V12.3514ZM14.2107 11.579V11.5791H1.78674L1.78643 4.76876H1.78647L1.78679 11.579H14.2107ZM2.07444 11.2914H13.923V5.05639H2.07416L2.07444 11.2914ZM14.7123 12.0638V4.28772H1.28767V4.28767H14.7123V12.0638H14.7123Z" fill="#1C1D21"/>
<path d="M12.7169 5.5H12.7332C13.0463 5.5 13.3 5.75735 13.3 6.07481V6.07556C13.3 6.39302 13.0463 6.65037 12.7332 6.65037H12.7169C12.4038 6.65037 12.15 6.39302 12.15 6.07556V6.07481C12.15 5.75735 12.4038 5.5 12.7169 5.5Z" fill="#1C1D21"/>
<path d="M3.57052 5.5H3.57951C3.8946 5.5 4.15002 5.75752 4.15002 6.07519C4.15002 6.39285 3.8946 6.65037 3.57951 6.65037H3.57052C3.25544 6.65037 3.00001 6.39285 3.00001 6.07519C3.00001 5.75752 3.25544 5.5 3.57052 5.5Z" fill="#1C1D21"/>
<path d="M12.7202 9.65039H12.7299C13.0448 9.65039 13.3 9.90791 13.3 10.2256C13.3 10.5432 13.0448 10.8008 12.7299 10.8008H12.7202C12.4053 10.8008 12.15 10.5432 12.15 10.2256C12.15 9.90791 12.4053 9.65039 12.7202 9.65039Z" fill="#1C1D21"/>
<path d="M3.5639 9.65039H3.58613C3.89756 9.65039 4.15002 9.90791 4.15002 10.2256C4.15002 10.5432 3.89756 10.8008 3.58613 10.8008H3.5639C3.25247 10.8008 3.00001 10.5432 3.00001 10.2256C3.00001 9.90791 3.25247 9.65039 3.5639 9.65039Z" fill="#1C1D21"/>
</svg>
