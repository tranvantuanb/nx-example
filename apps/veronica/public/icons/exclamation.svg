<svg width="15" height="15" viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.25 13.5C10.7018 13.5 13.5 10.7018 13.5 7.25C13.5 3.79822 10.7018 1 7.25 1C3.79822 1 1 3.79822 1 7.25C1 10.7018 3.79822 13.5 7.25 13.5Z" stroke="#1E40AF" stroke-width="1.5"/>
<path d="M7.25 10.0625V10.375M7.25 4.125V7.875V4.125Z" stroke="#1E40AF" stroke-width="1.5" stroke-linecap="round"/>
</svg>
