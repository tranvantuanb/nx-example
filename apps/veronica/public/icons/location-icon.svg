<svg width="12" height="14" viewBox="0 0 12 14" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M5.99995 7.6C6.99406 7.6 7.79995 6.79411 7.79995 5.8C7.79995 4.80589 6.99406 4 5.99995 4C5.00584 4 4.19995 4.80589 4.19995 5.8C4.19995 6.79411 5.00584 7.6 5.99995 7.6Z" stroke="#1C1D21" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M5.99995 1C4.72691 1 3.50601 1.50571 2.60584 2.40589C1.70566 3.30606 1.19995 4.52696 1.19995 5.8C1.19995 6.9352 1.44115 7.678 2.09995 8.5L5.99995 13L9.89995 8.5C10.5588 7.678 10.8 6.9352 10.8 5.8C10.8 4.52696 10.2942 3.30606 9.39406 2.40589C8.49389 1.50571 7.27299 1 5.99995 1V1Z" stroke="#1C1D21" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
