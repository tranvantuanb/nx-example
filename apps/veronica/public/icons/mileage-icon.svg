<svg width="13" height="10" viewBox="0 0 13 10" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M6.33334 0.5C2.84114 0.5 2.21015e-06 3.34113 2.21015e-06 6.83333C-0.000818853 7.8108 0.227138 8.7749 0.665636 9.6485C0.773302 9.86383 0.992436 10 1.23247 10H11.4336C11.6736 10 11.8934 9.86383 12.001 9.6485C12.4395 8.7749 12.6675 7.8108 12.6667 6.83333C12.6667 3.34113 9.82554 0.5 6.33334 0.5ZM11.0263 8.73333H1.64034C1.39285 8.13049 1.2659 7.485 1.26667 6.83333C1.26667 4.0397 3.5397 1.76667 6.33334 1.76667C9.12697 1.76667 11.4 4.0397 11.4 6.83333C11.4008 7.485 11.2738 8.13049 11.0263 8.73333Z" fill="#1C1D21"/>
<path d="M5.66143 6.16132C5.52671 6.24943 5.41605 6.36971 5.33947 6.5113C5.26288 6.65289 5.22278 6.81133 5.22278 6.97231C5.22278 7.13328 5.26288 7.29172 5.33947 7.43332C5.41605 7.57491 5.52671 7.69519 5.66143 7.78329C5.76789 7.85334 5.88711 7.9017 6.01228 7.9256C6.13745 7.94951 6.26611 7.94849 6.39089 7.9226C6.51566 7.89672 6.63411 7.84648 6.73944 7.77475C6.84477 7.70303 6.93492 7.61124 7.00473 7.50462L9.514 3.68056C9.51601 3.67721 9.51707 3.67338 9.51707 3.66947C9.51707 3.66557 9.51601 3.66174 9.514 3.65839C9.51101 3.65383 9.50636 3.65061 9.50104 3.64942C9.49571 3.64824 9.49013 3.64919 9.4855 3.65206L5.66143 6.16132Z" fill="#1C1D21"/>
</svg>
