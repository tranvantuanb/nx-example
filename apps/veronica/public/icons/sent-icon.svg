<svg width="64" height="64" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_1824_31955)">
<path d="M62.7 0.523199C62.2352 0.206437 61.691 0.0261105 61.129 0.00262755C60.567 -0.0208554 60.0096 0.113443 59.52 0.390324L1.51502 33.3875C0.523019 33.9625 -0.0629807 35.0374 0.00539428 36.1749C0.0737693 37.3156 0.784644 38.32 1.83789 38.7612L21.0004 46.8087V60.9962C21.0004 62.0938 21.6 63.1012 22.5629 63.6287C23.0125 63.875 23.5 64 23.8875 64C24.452 64 25.0173 63.8398 25.5113 63.5234L39.4113 54.591L51.7313 59.7659C52.0992 59.9205 52.4943 60.0001 52.8934 60.0003C53.4031 60.0003 53.911 59.8714 54.3659 59.6136C54.7643 59.3896 55.106 59.0773 55.365 58.7007C55.6239 58.3241 55.7931 57.8932 55.8596 57.4411L63.8596 3.44107C64.1375 2.30495 63.6375 1.17157 62.7 0.523199ZM46.1625 14.8999L22.775 41.0125L9.77877 35.5875L46.1625 14.8999ZM26.8875 55.5V49.33L32.6938 51.7687L26.8875 55.5ZM50.6 52.7374L28.525 43.4637L56.6 12.2762L50.6 52.7374Z" fill="#5E81F4"/>
</g>
<defs>
<clipPath id="clip0_1824_31955">
<rect width="64" height="64" fill="white"/>
</clipPath>
</defs>
</svg>
