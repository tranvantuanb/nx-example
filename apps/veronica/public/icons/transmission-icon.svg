<svg width="10" height="11" viewBox="0 0 10 11" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M9.52791 9.78771C10.1918 9.35706 9.8721 8.36047 9.07416 8.36047H8.3661C8.19893 8.36047 8.0636 8.49056 8.0636 8.65089V10.2836C8.0636 10.6686 8.66859 10.6667 8.66859 10.2836V9.92093H8.84005L9.35898 10.4773C9.62305 10.7604 10.0803 10.38 9.80969 10.0899L9.52791 9.78771ZM8.66861 8.94128H9.07416C9.34862 8.94128 9.34874 9.3401 9.07416 9.3401H8.66861V8.94128Z" fill="#1C1D21"/>
<path d="M1.28867 9.60773C2.12309 9.46091 2.01143 8.26868 1.16273 8.26868H0.30774C-0.0919748 8.26868 -0.0923915 8.84948 0.30774 8.84948H1.16273C1.27897 8.84948 1.29407 9.0163 1.17964 9.03645C-0.184066 9.27632 -0.198472 10.4285 0.30774 10.4285H1.56108C1.96079 10.4285 1.96121 9.84774 1.56108 9.84774H0.707952C0.851932 9.65673 1.02996 9.65324 1.28867 9.60773Z" fill="#1C1D21"/>
<path d="M0.477784 1.54247H0.632063V2.83999C0.632063 3.22497 1.23705 3.22325 1.23705 2.83999V1.25208C1.23705 1.09159 1.10153 0.96167 0.934548 0.96167H0.477764C0.0765809 0.96167 0.0788628 1.54247 0.477784 1.54247Z" fill="#1C1D21"/>
<path d="M5.17029 2.54073H4.32814C3.92843 2.54073 3.92801 3.12153 4.32814 3.12153H5.17029C5.74815 3.12153 6.08592 2.49312 5.7531 2.04161C6.08534 1.59086 5.74899 0.96167 5.17029 0.96167H4.32814C3.92843 0.96167 3.92801 1.54247 4.32814 1.54247H5.17029C5.31393 1.54247 5.31401 1.7512 5.17029 1.7512H4.65727C4.25756 1.7512 4.25714 2.332 4.65727 2.332H5.17029C5.31395 2.33202 5.31401 2.54073 5.17029 2.54073Z" fill="#1C1D21"/>
<path d="M5.68398 9.17657H5.50902V8.61302C5.50902 8.22929 4.90403 8.22889 4.90403 8.61302V9.17657H4.51075V8.61302C4.51075 8.22929 3.90576 8.22889 3.90576 8.61302V9.46697C3.90576 9.62735 4.04119 9.75738 4.20827 9.75738H4.90405V10.1921C4.90405 10.5758 5.50904 10.5762 5.50904 10.1921V9.75738H5.684C6.08369 9.75737 6.08411 9.17657 5.68398 9.17657Z" fill="#1C1D21"/>
<path d="M9.11274 2.54073H8.424C8.02428 2.54073 8.02387 3.12153 8.424 3.12153H9.11274C10.0589 3.12153 10.0592 1.7512 9.11274 1.7512H8.76422L8.78984 1.54247H9.52629C9.926 1.54247 9.92642 0.96167 9.52629 0.96167H8.52089C8.36751 0.96167 8.23843 1.07185 8.22049 1.21808L8.12358 2.00761C8.10239 2.18008 8.24291 2.332 8.42398 2.332H9.11272C9.25954 2.33202 9.2595 2.54073 9.11274 2.54073Z" fill="#1C1D21"/>
<path d="M8.66407 7.67893C8.66407 8.06266 9.26906 8.06306 9.26906 7.67893V3.8558C9.26906 3.47207 8.66407 3.47167 8.66407 3.8558V5.47696H6.10737C5.99827 5.07465 5.66773 4.75734 5.24867 4.65259V3.85571C5.24867 3.47209 4.64368 3.47144 4.64368 3.85571V4.65259C4.22462 4.75732 3.89409 5.07463 3.78498 5.47696H1.22828V3.8558C1.22828 3.47207 0.623291 3.47167 0.623291 3.8558V7.67893C0.623291 8.06266 1.22828 8.06306 1.22828 7.67893V6.05778H3.78498C3.89407 6.46008 4.22462 6.77739 4.64368 6.88214V7.67902C4.64368 8.06264 5.24867 8.06329 5.24867 7.67902V6.88214C5.66773 6.77741 5.99825 6.46008 6.10737 6.05778H8.66407V7.67893ZM5.54109 5.76737C5.54109 6.08222 5.27423 6.33849 4.94618 6.33849C4.61822 6.33849 4.35127 6.08229 4.35127 5.76737C4.35127 5.45269 4.61796 5.19624 4.94618 5.19624C5.27397 5.19624 5.54109 5.45227 5.54109 5.76737Z" fill="#1C1D21"/>
</svg>
