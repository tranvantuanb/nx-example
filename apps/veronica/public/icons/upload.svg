<svg width="13" height="12" viewBox="0 0 13 12" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M4.81226 2.96905H5.96694V8.25499C5.96694 8.32374 6.02319 8.37999 6.09194 8.37999H7.02944C7.09819 8.37999 7.15444 8.32374 7.15444 8.25499V2.96905H8.31226C8.41694 2.96905 8.47476 2.84874 8.41069 2.76749L6.66069 0.551867C6.649 0.536927 6.63406 0.524844 6.61701 0.516534C6.59995 0.508225 6.58123 0.503906 6.56226 0.503906C6.54328 0.503906 6.52456 0.508225 6.50751 0.516534C6.49045 0.524844 6.47551 0.536927 6.46382 0.551867L4.71382 2.76593C4.64976 2.84874 4.70757 2.96905 4.81226 2.96905ZM12.281 7.78624H11.3435C11.2748 7.78624 11.2185 7.84249 11.2185 7.91124V10.3175H1.90601V7.91124C1.90601 7.84249 1.84976 7.78624 1.78101 7.78624H0.843506C0.774756 7.78624 0.718506 7.84249 0.718506 7.91124V11.005C0.718506 11.2816 0.941943 11.505 1.21851 11.505H11.906C12.1826 11.505 12.406 11.2816 12.406 11.005V7.91124C12.406 7.84249 12.3498 7.78624 12.281 7.78624Z" fill="black" fill-opacity="0.85"/>
</svg>
