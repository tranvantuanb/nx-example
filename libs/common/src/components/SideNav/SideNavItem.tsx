import React, { memo } from 'react';
import tw, { styled } from 'twin.macro';
import Link from 'next/link';
import { TopNavItemProps } from './SideNav';

interface Props {
  navigationItem: TopNavItemProps;
  isActive: boolean;
  isExpanded: boolean;
}

interface StyledProps {
  isActive: boolean;
  isExpanded?: boolean;
}

const StyledIcon = styled.span<StyledProps>`
  ${tw`text-[19px] h-8 w-6 rounded-lg grid place-content-center`}

  ${({ isActive }) =>
    isActive
      ? tw`text-indigo-500 text-[19px]`
      : tw`text-gray-400 group-hover:text-gray-500`}
`;

const StyledNavContainer = styled.div`
  ${tw`bg-white flex justify-between`}
`;

const StyledNavLink = styled.div<StyledProps>`
  ${tw`flex items-center px-3 py-2 text-base rounded-md`}
  ${tw`transition-all	ease-out`}
  ${tw`hover:cursor-pointer`}

  ${({ isActive }) =>
    isActive ? tw`bg-blue-50` : tw`bg-white hover:bg-gray-100`}

  ${({ isExpanded }) => (isExpanded ? tw`w-[202px]` : tw`w-[48px]`)}
`;

const StyledHighlight = styled.div`
  ${tw`bg-indigo-500 w-[2px]`}
`;

const StyledTitle = styled.p<StyledProps>`
  ${tw`ml-2 text-[14px] select-none whitespace-nowrap`}

  ${({ isActive }) => (isActive ? tw`text-black-base` : tw`text-gray-500`)}
`;

export const SideNavItem: React.FC<Props> = memo(
  ({ navigationItem, isActive = false, isExpanded = true }) => {
    const { name, href } = navigationItem;
    const Icon = navigationItem.icon;

    return (
      <StyledNavContainer data-testid="nav-item">
        <div className="flex justify-center w-full">
          <Link href={href}>
            <StyledNavLink
              className="nav-link"
              isExpanded={isExpanded}
              isActive={isActive}
            >
              <StyledIcon className="group" isActive={isActive}>
                <Icon />
              </StyledIcon>
              {isExpanded && (
                <StyledTitle isActive={isActive}>{name}</StyledTitle>
              )}
            </StyledNavLink>
          </Link>
        </div>
        {isActive && <StyledHighlight />}
      </StyledNavContainer>
    );
  }
);
