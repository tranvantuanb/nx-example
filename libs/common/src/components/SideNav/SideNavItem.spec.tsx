import { HomeOutlined } from '@ant-design/icons';
import { render, screen } from '../../test';

import { SideNavItem } from './SideNavItem';

const NAV_ITEM = {
  name: 'Home',
  href: '/',
  pathname: '/',
  icon: HomeOutlined,
};

describe('SideNav', () => {
  it('should render successfully with icon only', async () => {
    const { container, getByRole } = render(
      <SideNavItem
        navigationItem={NAV_ITEM}
        isExpanded={false}
        isActive={false}
      />
    );
    expect(container).toBeTruthy();

    expect(screen.queryByText('Home')).toBeFalsy();
    const navItemIcon = getByRole('img');
    expect(navItemIcon).toBeTruthy();
  });

  it('should render expanded nav item with text and icon', async () => {
    const { container, getByRole } = render(
      <SideNavItem navigationItem={NAV_ITEM} isExpanded isActive={false} />
    );
    expect(container).toBeTruthy();

    expect(screen.getByText('Home')).toBeTruthy();
    const navItemIcon = getByRole('img');
    expect(navItemIcon).toBeTruthy();
  });

  it('should render active nav item with background color', async () => {
    const { container } = render(
      <SideNavItem navigationItem={NAV_ITEM} isExpanded isActive />
    );
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
