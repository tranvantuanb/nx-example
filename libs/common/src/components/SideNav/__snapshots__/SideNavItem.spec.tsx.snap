// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should render active nav item with background color 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

<div>
  <div
    class="css-13yb8hl-StyledNavContainer eagsyzi3"
    data-testid="nav-item"
  >
    <div
      class="c0"
    >
      <div
        class="nav-link css-1xfujy4-StyledNavLink eagsyzi2"
      >
        <span
          class="group css-1as126y-StyledIcon eagsyzi4"
        >
          <span
            aria-label="home"
            class="anticon anticon-home"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
        </span>
        <p
          class="css-196ekmk-StyledTitle eagsyzi0"
        >
          Home
        </p>
      </div>
    </div>
    <div
      class="css-1m2n0g9-StyledHighlight eagsyzi1"
    />
  </div>
</div>
`;
